apps/remote_control/lib/lexical/remote_control/search/fuzzy/scorer.ex

```diff
@@ -150,36 +163,52 @@ defmodule Lexical.RemoteControl.Search.Fuzzy.Scorer do
     @non_match_score
   end
 
-  defp calculate_score(%__MODULE__{} = score, subject() = subject, pattern) do
+  defp calculate_score(%__MODULE__{} = score, subject(graphemes: graphemes) = subject, pattern) do
     pattern_length = String.length(pattern)
 
     {consecutive_count, consecutive_bonus} =
-      consecutive_match_bonus(score.matched_character_positions)
+      consecutive_match_boost(score.matched_character_positions)
 
-    match_amount_boost = consecutive_count * pattern_length * 10
+    match_amount_boost = consecutive_count * pattern_length
 
-    [first_match_position | _] = score.matched_character_positions
+    match_boost = match_boost(score, subject)
 
-    pattern_length_boost = pattern_length
-
-    # penalize first matches further in the string by making them negative.
-    first_match_bonus = max(0 - first_match_position, 10)
-
-    case_match_boost = case_match_boost(pattern, score.matched_character_positions, subject)
+    camel_case_boost = camel_case_boost(score.matched_character_positions, subject)
 
     mismatched_penalty = mismatched_penalty(score.matched_character_positions)
 
-    pattern_length_boost + consecutive_bonus + first_match_bonus + case_match_boost +
-      match_amount_boost - mismatched_penalty
+    incompleteness_penalty = tuple_size(graphemes) - length(score.matched_character_positions)
+
+    consecutive_bonus + match_boost + camel_case_boost +
+      match_amount_boost - mismatched_penalty - incompleteness_penalty
   end
 
   defp normalize(string) do
     String.downcase(string)
   end
 
+  @last_period_boost 15
+  @max_match_boost_boost 10
+  defp match_boost(%__MODULE__{} = score, subject(last_period_position: nil)) do
+    # penalize first matches further in the string by making them negative.
+    [first_match_position | _] = score.matched_character_positions
+    max(0 - first_match_position, @max_match_boost_boost)
```

Comment:
Won't this always return `@max_match_boost_boost`, since this is essentially `max(something_negative, 10)` and `10` will always be greater? Is this meant to be `max(-first_match_position, -@max_match_boost_boost)`?

There's also an instance of this below.

Reply:
Looking at the code, I'm wondering if this code should just always return `@max_match_boost_boost` (since it's a subject without a period, which is like the last segment matching). And then the clause below should maybe just return 0 because it doesn't meet the criteria for the boost?

Reply:
Yes good catch.